test/compilable/test16607.d

```diff
@@ -0,0 +1,15 @@
+struct A(T)
+{
+    T t; // causes A to be SIZEOKfwd b/c B (passed as T) isn't yet done
```

Comment:
Thanks for raising this, getting understood is important, and this is an useful consideration.